Add Auth0 docs (#3249)
hashicorp/web-unified-docs: content/vault/v2.x/content/ai/iam/providers/auth0.mdx

   identifier value becomes the JWT `aud` claim.

1. For **JSON Web Token (JWT) Profile**, select **RFC 9068**.

1. Scroll down and click **Create**.

1. Click the **Settings**.

1. Scroll down to **RBAC Settings**.

1. Click the toggle to switch on **Enable RBAC**.

1. Click the toggle to switch on **Add Permissions in the Access Token**.

1. Click **Save**.

## Step 3: Create an application

1. Navigate to **Applications** > **Applications**, then click
   **Create Application**.

1. In the **Name** field, enter the application name and select the application type.

1. Click **Create**.

1. Click **Settings**. You can find the **Client ID** and **Client Secret** of
   your application.

1. Scroll to **Application URIs** section. In the **Allowed Callback URLs**
   field, enter the `oauth2c` callback URL.

1. Scroll to the **Authorization Requests** section and turn **ON**
   **Require Pushed Authorization Requests (PAR)** so PAR is mandatory for this
   application.

1. Click **Advanced Settings**.

1. Click **OAuth**.

1. In the **Allowed APPs/APIs** field, enter the Vault API address (for example,
   `https://<vault-addr>:8200/v1`)

> 1. Open the **Grant Types** tab and ensure **Authorization Code** is checked.

Reviewer: ```suggestion
1. Open the **Grant Types** tab and check **Authorization Code**.
```

Management API. Find the grant id with
`GET /api/v2/client-grants?client_id=${APP_CLIENT_ID}`, then patch it:

```shell-session
$ curl -X PATCH "https://${APP_DOMAIN}/api/v2/client-grants/<GRANT_ID>" \
    -H "Authorization: Bearer ${MGMT_TOKEN}" \
    -H "Content-Type: application/json" \
    -d '{"authorization_details_types":["vault:path_access"]}'
```

The `authorization_details_types` field applies only to user-delegated access
flows.

</Tab>
</Tabs>

## Step 7: Create an agent user

1. Navigate to **User Management** > **Users**.

1. Click **Create User** and select **Create via UI** or **Create via JSON**.

1. Enter the **Email** and **Password**.

1. Click **Create**.

1. On the user page, click **Edit** next to **Email**, and set the email as
   verified.

## Step 8: Capture a baseline token

Export the Auth0 values you recorded, then run `oauth2c` to confirm the trust
and to capture the `iss` and `sub` claims you need for the Vault profile and
entity aliases.

1. Save the Auth0 configuration values as environment variables. Replace the
   placeholders with your actual values from the Auth0 application settings.

    ```shell-session

>     $ export APP_AUTH_URL="https://YOUR_TENANT.us.auth0.com/authorize" \

Reviewer: ```suggestion
    $ export APP_AUTH_URL="https://<your_tenant>.us.auth0.com/authorize" \
```

>         APP_CLIENT_ID="<application client id>" \

Reviewer: ```suggestion
        APP_CLIENT_ID="<application_client_id>" \
```

>         APP_CLIENT_SECRET="<application client secret>" \

Reviewer: ```suggestion
        APP_CLIENT_SECRET="<application_client_secret>" \
```

>         APP_JWKS_URI="https://YOUR_TENANT.us.auth0.com/.well-known/jwks.json"

Reviewer: ```suggestion
        APP_JWKS_URI="https://<your_tenant>.us.auth0.com/.well-known/jwks.json"
```

    <CodeBlockConfig hideClipboard>

    ```plaintext
    SUCCESS  Exchanged authorization code for access token
    ```

    </CodeBlockConfig>

    Example token values:

    <CodeBlockConfig hideClipboard>

    ```json
    {
      "aud": "https://localhost:8200/v1",
      "client_id": "EBYeLHzuLqSeyUYLY7EC3hqRgvaiS9Dv",
      "exp": 1782402664,
      "iat": 1782316264,
      "iss": "https://hashicups-dev.us.auth0.com/",
      "jti": "baFC3mowpruhSr1JdDRhbJ",
      "permissions": [],
      "sub": "auth0|c0ff33a2dc1f8e9de0dd95d7"
    }
    ```

    </CodeBlockConfig>

1. Record the `iss` and `sub` values and export them. The `sub` value contains
   a pipe character, so wrap it in single quotes so the shell doesn't use it.

    For example:

    ```shell-session
    $ export APP_ISS_URL="https://hashicups-dev.us.auth0.com/" \
        APP_JWT_SUB='auth0|c0ff33a2dc1f8e9de0dd95d7'
    ```

    This `sub` is the agent's external identity. In the **create the subject and
    actor entities and aliases** step you will create a separate external
    identity for Danielle the human subject.

## Step 9: Save your provider information

> Confirm your subject and actor details from Verify. To configure agentic IAM,

Reviewer: ```suggestion
Confirm your subject and actor details from Auth0. To configure agentic IAM,
```

you must have the following information:

| Value             | Environmental variable | Description                                                                                       |
| ----------------- | ---------------------- | ------------------------------------------------------------------------------------------------- |
| Authorization URL | `APP_AUTH_URL`         | Authorization URL of the application: `https://YOUR_TENANT.us.auth0.com/authorize`                |

> | Issur URL         | `APP_ISS_URL`          | URL identifying the authentication entry point: `https://APP_NAME.us.auth0.com/                   |

Reviewer: ```suggestion
| Issuer URL         | `APP_ISS_URL`          | URL identifying the authentication entry point: `https://APP_NAME.us.auth0.com/`                   |
```

| JWT subject       | `APP_JWT_SUB`          | The JWT claim to use as the user identifier                                                       |
| JWKS URL          | `APP_JWKS_URI`         | URI to fetch public keys in JWKS format: `https://YOUR_TENANT.us.auth0.com/.well-known/jwks.json` |

> | Clinet ID         | `APP_CLIENT_ID`        | Client ID of the application you registered in Auth0                                              |

Reviewer: ```suggestion
| Client ID         | `APP_CLIENT_ID`        | Client ID of the application you registered in Auth0                                              |
```

| Client secret     | `APP_CLIENT_SECRET`    | Client secret associated with the application                                                     |
| Audience address  | `AUDIENCE_ADDR`        | Vault cluster address including the API version, `/v1`                                            |
| Domain address    | `APP_DOMAIN`           | Domain of the application you registered in Auth0                                                 |

Set the following environment variables to make Vault configuration more
convenient:

```shell-session
$ export VAULT_ADDR="<your_vault_server_url>" \
      APP_AUTH_URL="<app_auth_url>" \
      APP_CLIENT_ID="<app_client_id>" \
      APP_CLIENT_SECRET="<app_client_secret>" \
      APP_JWKS_URI="<app_jwks_url>" \

>       APP_JWKS_SUB="<app_jwks_sub>" \

Reviewer: ```suggestion
      APP_JWT_SUB="<app_jwt_sub>" \
```

Reviewer: Updating to match the name from the table.

      APP_ISS_URL="<app_iss_url>" \
      AUDIENCE_ADDR="<aud_address>" \
      APP_DOMAIN="<app_domain>"       
```

For example:

<CodeBlockConfig hideClipboard>

```shell-session
$ export VAULT_ADDR="https://127.0.0.1:8200" \
      APP_AUTH_URL="https://abc-123if67e.us.auth0.com/authorize" \
      APP_CLIENT_ID="ABCD...abc" \
      APP_CLIENT_SECRET="000000...000000" \
      APP_JWKS_URI="https://abc-123if67e.us.auth0.com/.well-known/jwks.json" \

>       APP_JWKS_SUB="auth0|c0ff33a2dc1f8e9de0dd95d7" \

Reviewer: ```suggestion
      APP_JWT_SUB="auth0|c0ff33a2dc1f8e9de0dd95d7" \
```

Reviewer: Updating to match the name from the table.

      APP_ISS_URL="https://hashicups-dev.us.auth0.com/" \
      AUDIENCE_ADDR="https://127.0.0.1:8200/v1" \
      APP_DOMAIN="abc-123if67e.auth0.com"      
```

</CodeBlockConfig>

## Next steps

- Configure Vault to [secure agentic workflows](/vault/ai/iam/setup)

> - Configure additonal identity provides and authentication servers:

Reviewer: ```suggestion
- Configure additional identity providers and authentication servers:
```
